src/dialogs/restart/dialog-restart-wait.ts

```diff
@@ -0,0 +1,175 @@
+import { haStyle, haStyleDialog } from "../../resources/styles";
+import type { HomeAssistant } from "../../types";
+import type { RestartWaitDialogParams } from "./show-dialog-restart";
+
+@customElement("dialog-restart-wait")
+class DialogRestartWait extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _open = false;
+
+  @state()
+  private _title = "";
+
+  private _actionOnIdle?: () => Promise<void>;
+
+  @state()
+  private _error?: string;
+
+  @state()
+  private _backupState?: ManagerState;
+
+  private _backupEventsSubscription?: UnsubscribeFunc;
+
+  @query("ha-md-dialog") private _dialog?: HaMdDialog;
+
+  public async showDialog(params: RestartWaitDialogParams): Promise<void> {
+    this._open = true;
+    this._loadBackupState();
+
+    this._title = params.title;
+    this._backupState = params.initialBackupState;
+
+    this._actionOnIdle = params.action;
+  }
+
+  private _dialogClosed(): void {
+    this._open = false;
+
+    if (this._backupEventsSubscription) {
+      this._backupEventsSubscription();
+      this._backupEventsSubscription = undefined;
+    }
+
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  public closeDialog(): void {
+    this._dialog?.close();
+  }
+
+  private _getWaitMessage() {
+    switch (this._backupState) {
+      case "create_backup":
+        return this.hass.localize("ui.dialogs.restart.wait_for_backup");
+      case "receive_backup":
+        return this.hass.localize("ui.dialogs.restart.wait_for_upload");
+      case "restore_backup":
+        return this.hass.localize("ui.dialogs.restart.wait_for_restore");
+      default:
+        return "";
+    }
+  }
+
+  protected render() {
+    if (!this._open) {
+      return nothing;
+    }
+
+    const waitMessage = this._getWaitMessage();
+
+    return html`
+      <ha-md-dialog
+        open
+        @closed=${this._dialogClosed}
+        .disableCancelAction=${true}
+      >
+        <ha-dialog-header slot="headline">
+          <ha-icon-button
+            slot="navigationIcon"
+            .label=${this.hass.localize("ui.common.cancel")}
+            .path=${mdiClose}
+            @click=${this.closeDialog}
+          ></ha-icon-button>
+          <span slot="title" .title=${this._title}> ${this._title} </span>
+        </ha-dialog-header>
+        <div slot="content" class="content">
+          ${this._error
+            ? html`<ha-alert alert-type="error"
+                >${this.hass.localize("ui.dialogs.restart.error_backup_state", {
+                  error: this._error,
+                })}</ha-alert
+              > `
+            : html`
+                <ha-spinner></ha-spinner>
+                ${waitMessage}
+              `}
+        </div>
+      </ha-md-dialog>
+    `;
+  }
+
+  private async _loadBackupState() {
+    try {
+      const { state: backupState } = await fetchBackupInfo(this.hass);
+      this._backupState = backupState;
```

Comment:
Why do we load the backupstate again when we already got it through the params of the dialog and are going to subscribe? We can just subscribe and use that state?

Reply:
I thought I don't get an initial event and when it sets to idle in the meantime I need to know, but it seems that I always get the current state as initial event :+1: